terraform/services/bucket-access-log/main.tf

```diff
@@ -0,0 +1,53 @@
+resource "aws_s3_bucket" "bucket-access_logs" {
+  bucket        = "${var.app}-${var.env}-bucket-access-log"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_versioning" "bucket-access_logs" {
+  bucket = aws_s3_bucket.bucket-access_logs.id
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "bucket-access_logs" {
+  bucket = aws_s3_bucket.bucket-access_logs.id
+
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "aws:kms"
+    }
+  }
+}
+
+data "aws_iam_policy_document" "bucket-access_logs" {
```

Comment:
Adding this policy to bucket access logs. I have a worry. this part of the policy that's the bucket access logs' name and the path/prefix to the logs
resources = ["arn:aws:s3:::${local.bucket}/${local.s3_logs_prefix}/*"]
my worry is that in the bucket module we are using the bucket name as prefix to logs that is var.name, so am not sure how it will work. I used data source to look up the bucket name in the bucket service but that means the bucket in the module has to be created first for data look up to work and we are also using a data source to look up the access log bucket in the the bucket module